src/Domain/Insights/Sniff.php

```diff
@@ -80,4 +80,17 @@ public function getDetails(): array
             return $error->getFileInfo()->getRealPath() . ':' . $error->getLine() . ': ' . $error->getMessage();
         }, $this->errors);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getInsightClass(): string
+    {
+       return $this->getSniffClass($this->errors[0]);
```

Comment:
Let's play defensive here, can you create an exception like `SniffClassNotFound` and throw it when there is no errors?

Reply:
:+1: